src/odemis/acq/test/stream_test.py

```diff
@@ -3292,12 +3309,130 @@ def test_spec_calib(self):
+        self.assertEqual(im2d.shape, spec.shape[-2:] + (3,))
+
+        # Create projections
+        proj_point_spectrum = SinglePointSpectrumProjection(specs)
+        proj_point_chrono = SinglePointChronoProjection(specs)
+
+        # Test
+        tl = spec.metadata.get(model.MD_TIME_LIST)
+        wl = spec.metadata.get(model.MD_WL_LIST)
+
+        for time_index in range(0, 3):
+            for wl_index in range(0, 3):
+                for x in range(100, 10):
+                    for y in range(100, 10):
+                        specs.selected_pixel.value = (x, y)
+                        specs.selected_time.value = tl[time_index]
+                        specs.selected_wavelength.value = wl[wl_index]
+                        time.sleep(0.2)
+                        self.assertListEqual(proj_point_spectrum.image.value.tolist(), spec[:, :, 0, y, x].tolist())
+                        self.assertListEqual(proj_point_chrono.image.value.tolist(), spec[wl_index, :, 0, y, x].tolist())
+
+    def test_temp_spec_calib(self):
+        """Test StaticSpectrumStream calibration"""
+        spec = self._create_temporal_spec_data()
+        specs = stream.StaticSpectrumStream("test", spec)
+        proj_spatial = RGBSpatialSpectrumProjection(specs)
+        specs.spectrumBandwidth.value = (specs.spectrumBandwidth.range[0][0], specs.spectrumBandwidth.range[1][1])
+        time.sleep(0.5)  # ensure that .image is updated
+
+        # Check efficiency compensation
+        prev_im2d = proj_spatial.image.value
+
+        dbckg = numpy.ones(spec.shape, dtype=numpy.uint16) + 10
+        wl_bckg = list(spec.metadata[model.MD_WL_LIST])
+        obckg = model.DataArray(dbckg, metadata={model.MD_WL_LIST: wl_bckg})
+        bckg = calibration.get_spectrum_data([obckg])
+
+        dcalib = numpy.array([1, 1.3, 2, 3.5, 4, 5, 1.3, 6, 9.1], dtype=numpy.float)
+        dcalib.shape = (dcalib.shape[0], 1, 1, 1, 1)
+        wl_calib = 400e-9 + numpy.array(range(dcalib.shape[0])) * 10e-9
+        calib = model.DataArray(dcalib, metadata={model.MD_WL_LIST: wl_calib})
+
+        specs.efficiencyCompensation.value = calib
+
+        specs.background.value = bckg
+
+        # Control spatial spectrum
+        time.sleep(0.5)
+        im2d = proj_spatial.image.value
+        # Check it's a RGB DataArray
+        self.assertEqual(im2d.shape, spec.shape[-2:] + (3,))
+        self.assertTrue(numpy.any(im2d != prev_im2d))
+
+    def test_mean_spec(self):
+        spec = self._create_spec_data()
+        specs = stream.StaticSpectrumStream("test", spec)
+        proj = MeanSpectrumProjection(specs)
+        time.sleep(2)
+        mean_spec = proj.image.value
+        self.assertEqual(mean_spec.shape, (spec.shape[0],))
+
+    def test_mean_temporal_spec(self):
+        temp_spec = self._create_temporal_spec_data()
+        temp_specs = stream.StaticSpectrumStream("test", temp_spec)
+        proj = MeanSpectrumProjection(temp_specs)
+        time.sleep(5)
```

Comment:
Why does it need 5s? On one billion pixels, on my computer, the mean only takes 0.6s. So I don't expect any time longer that 1s here. (The histogram seems to take quite a lot of time, 5s for 1 billion pixels, but it should be running in a separate thread).  With the smaller shape, you should be able to keep it under 20ms, and histogram under 0.1s, so a sleep of 1s should be enough.